src/compiler/binder.ts

```diff
@@ -2416,7 +2416,7 @@ namespace ts {
 
             // 'module.exports = expr' assignment
             const flags = exportAssignmentIsAlias(node)
-                ? SymbolFlags.Alias // An export= with an EntityNameExpression or a ClassExpression exports all meanings of that identifier or class
+                ? SymbolFlags.Alias | SymbolFlags.Assignment // An export= with an EntityNameExpression or a ClassExpression exports all meanings of that identifier or class

```

Comment:
So why is it not flagged as Assignment in the other case?

Reply:
No good reason! The other uses of Assignment are unconditional now as well. Nice catch.